### ISSUE TITLE
Error 500 supporting proposals as a manager

### ISSUE THREAD

**Issue by javierv** [consul/consul#2767]
## Steps to reproduce

1. Log in as manager.
1. Enter the management area.
1. Click on "Support proposals".
1. Click the "Support" link on any proposal.

## Expected Behavior

The number of supports should increase and the message "You have already supported this proposal. Share it!" should replace the "Support" link.

## Actual Behavior

Nothing happens. The rails server reports an error in the AJAX request:

```
Completed 500 Internal Server Error in 726ms (ActiveRecord: 2.5ms)

ActionView::Template::Error (undefined local variable or method `vote_url' for #<#<Class:0x007f3edf77de48>:0x007f3edf4329c8>
Did you mean?  voted_for?):
    21:         <%= t("proposals.proposal.already_supported") %>
    22:       </div>
    23:     <% elsif user_signed_in? && proposal.votable_by?(current_user) %>
    24:       <%= link_to vote_url,
    25:           class: "button button-support small expanded",
    26:           title: t('proposals.proposal.support_title'), method: "post", remote: true do %>
    27:         <%= t("proposals.proposal.support") %>
  app/views/proposals/_votes.html.erb:24:in `_app_views_proposals__votes_html_erb__1078875210924081062_47229820199620'
  app/views/proposals/vote.js.erb:1:in `_app_views_proposals_vote_js_erb___58443929774069076_47229820638400'
  app/views/management/proposals/vote.js.erb:1:in `_app_views_management_proposals_vote_js_erb___514855472135197839_47229797291940'
```

## Notes

There's a reason while the related test (`spec/features/management/proposals_spec.rb:140`: "Proposals Voting Voting proposals on behalf of someone in index view") passes while the same code crashes during development.

The code assigning the votes is:

```ruby
# Management::ProposalsController
def vote
  @proposal.register_vote(managed_user, 'yes')
  set_proposal_votes(@proposal)
end

def set_proposal_votes(proposals)
  @proposal_votes = managed_user ? managed_user.proposal_votes(proposals) : {}
end
```

During development, `managed_user` has `nil` as id, as shown in a debugging session:

```ruby
(byebug) managed_user
#<User id: nil, email: "", encrypted_password: ""
# ...
(byebug) @proposal.register_vote(managed_user, 'yes')
   (0.7ms)  SELECT "settings"."value" FROM "settings" WHERE "settings"."key" = $1  ORDER BY "settings"."id" ASC  [["key", "feature.user.skip_verification"]]
   (0.4ms)  SELECT "settings"."value" FROM "settings" WHERE "settings"."key" = $1  ORDER BY "settings"."id" ASC  [["key", "months_to_archive_proposals"]]
   (0.4ms)  SELECT COUNT(*) FROM "votes" WHERE "votes"."votable_id" = $1 AND "votes"."votable_type" = $2 AND "votes"."voter_type" = $3 AND "votes"."voter_id" IS NULL AND "votes"."vote_scope" IS NULL  [["votable_id", 37], ["votable_type", "Proposal"], ["voter_type", "User"]]
   (0.3ms)  BEGIN
   (0.3ms)  ROLLBACK
false
```

However, the test manually stubs the `managed_user` method:

```ruby
# In the related spec:
user = create(:user, :level_two)
login_managed_user(user)

# In the support file:
def login_managed_user(user)
  allow_any_instance_of(Management::BaseController).to receive(:managed_user).and_return(user)
end

```

I can only confirm this behaviour on my machine. I'm not sure about production or preproduction issues.

**Comment by javierv** [comment]
I think I'd need some help on this one. Tracing the code, we find:

```ruby
# Management::BaseController.
def managed_user
  @managed_user ||= Verification::Management::ManagedUser.find(session[:document_type], session[:document_number])
end

# Management::DocumentVerificationsController
before_action :set_document, only: :check

def set_document
  session[:document_type] = params[:document_verification][:document_type]
  session[:document_number] = params[:document_verification][:document_number]
  clear_password
end
```

But `DocumentVerificationsController#check` isn't accessed during the whole proccess, `session[:document_type]` is never set, `Verification::Management::ManagedUser.find` always returns a user with `nil` as ID, and the vote cannot be registered.

And here's where I'm a bit lost: I haven't understood how `managed_user` is supposed to work :confused:. Maybe the changes in f55c397 are related.

**Comment by javierm** [comment]
Closed via #4443.
